voice/voice-ai/x/Actions/ActionsView.swift

```diff
@@ -197,21 +199,32 @@ struct ActionsView: View {
     func handleOtherActions(actionType: ActionType) {
         switch actionType {
         case .reset:
-            SpeechRecognition.shared.reset()
+            DeepgramASR.shared.reset()
+            
+            DispatchQueue.main.asyncAfter(deadline: .now() + 0.2) {
+                DeepgramASR.shared.setup()
+            }
+            break;
         case .skip:
             stopRecording()
+            break;
         case .randomFact:
-            SpeechRecognition.shared.randomFacts()
+            DeepgramASR.shared.makeQuery("Give me one random fact")
+            break;
         case .play:
-            if SpeechRecognition.shared.isPaused() {
-                SpeechRecognition.shared.continueSpeech()
+            if isPlaying {
+                DeepgramASR.shared.continueSpeech()
             } else {
-                SpeechRecognition.shared.pause()
+                DeepgramASR.shared.pause()
             }
+            self.isPlaying.toggle()
+            break;
         case .repeatLast:
-            SpeechRecognition.shared.repeate()
+            DeepgramASR.shared.repeate()
+            break;
         case .speak:
-            stopRecording()
```

Comment:
Since we are making speaking / listening mutually exclusive right now, I think it should call `cut` - which immediately make the synthesizer "finish" thereby activate the listening mode again (through the `didFinish` version of the `AVSpeechSynthesizerDelegate`), and allow the user to speak